test/XmlHttpRequestConnection.tests.js

```diff
@@ -139,6 +139,40 @@ define(function (require) {
                 expect(mockCallback.mostRecentCall.args[1]).toEqual(jasmine.any(Response));
             });
 
+            it("execute call with custom http verb that needs to use POST", function (){
+
+                mockRequest.method = "PUBLISH";
+
+                connection.execute(
+                    mockRequest,
+                    mockCallback
+                );
+
+                expect(mockXMLHttpRequest.prototype.open).toHaveBeenCalled();
```

Comment:
I know the others tests are not written like that... but this can be simplified a bit with:

```js
expect(mockXMLHttpRequest.prototype.open).toHaveBeenCalledWith("POST", "/", true, "login", "password");                expect(mockXMLHttpRequest.prototype.setRequestHeader).toHaveBeenCalledWith("testHeader", "testHeaderValue");
expect(mockXMLHttpRequest.prototype.setRequestHeader).toHaveBeenCalledWith("X-HTTP-Method-Override", "PUBLISH");                expect(mockXMLHttpRequest.prototype.setRequestHeader.calls.length).toEqual(2);
expect(mockXMLHttpRequest.prototype.send).toHaveBeenCalledWith({testBody: ""});
```